scripts/ccpp_track_variables.py

```diff
@@ -0,0 +1,220 @@
+
+parser = argparse.ArgumentParser()
+parser.add_argument('-s', '--sdf',           action='store', \
+                    help='suite definition file to parse', required=True)
+parser.add_argument('-m', '--metadata_path', action='store', \
+                    help='path to CCPP scheme metadata files', required=True)
+parser.add_argument('-c', '--config',        action='store', \
+                    help='path to CCPP prebuild configuration file', required=True)
+parser.add_argument('-v', '--variable',      action='store', \
+                    help='variable to track through CCPP suite', required=True)
+#parser.add_argument('--draw',  action='store_true', \
+#                    help='draw graph of calling tree for given variable', default=False)
+parser.add_argument('--debug', action='store_true', help='enable debugging output', default=False)
+args = parser.parse_args()
+
+###############################################################################
+# Functions and subroutines                                                   #
+###############################################################################
+
+def parse_arguments(args):
+    """Parse command line arguments."""
+    success = True
+    sdf = args.sdf
+    var = args.variable
+    configfile = args.config
+    metapath = args.metadata_path
+    debug = args.debug
+    return(success,sdf,var,configfile,metapath,debug)
+
+def setup_logging(debug):
+    """Sets up the logging module and logging level."""
+    success = True
+    if debug:
+        level = logging.DEBUG
+    else:
+        level = logging.WARNING
+    logging.basicConfig(format='%(levelname)s: %(message)s', level=level)
+    if debug:
+        logging.info('Logging level set to DEBUG')
+    return success
+
+def parse_suite(sdf):
+    """Reads the provided sdf, parses into a Suite data structure, including the "call tree": 
+       the ordered list of schemes for the suite specified by the provided sdf"""
+    logging.info(f'Reading sdf {sdf} and populating Suite object')
+    suite = Suite(sdf_name=sdf)
+    success = suite.parse()
+    if not success:
+        logging.error(f'Parsing suite definition file {sdf} failed.')
+        success = False
+        return (success, suite)
+    logging.info(f'Successfully read sdf {suite.sdf_name}')
+    logging.info(f'Creating calling tree of schemes for suite {suite.name}')
+    success = suite.make_call_tree()
+    if not success:
+        logging.error('Parsing suite definition file {0} failed.'.format(sdf))
+        success = False
+    return (success, suite)
+
+def create_metadata_filename_dict(metapath):
+    """Given a path, read all .meta files in that directory and add them to a dictionary: the keys
+       are the name of the scheme, and the values are the filename of the .meta file associated
+       with that scheme"""
+
+    success = True
+    metadata_dict = {}
+    scheme_filenames=glob.glob(metapath + "*.meta")
```

Comment:
I'm not sure this will work if you don't explicitly have a `/` at the end of your path provided in command line arguments. A more robust solution might be:

```
scheme_filenames=glob.glob(os.path.join(metapath, "*.meta"))
```